data/transactions/logic/assembler.go

```diff
@@ -827,14 +831,38 @@ func asmBranch(ops *OpStream, spec *OpSpec, args []string) error {
 		return ops.error("branch operation needs label argument")
 	}
 
-	ops.referToLabel(ops.pending.Len(), args[0])
 	ops.pending.WriteByte(spec.Opcode)
+	ops.referToLabel(ops.pending.Len(), args[0], ops.pending.Len()+3)
```

Comment:
Perhaps `spec.Size` instead of `3` would be more self-documenting here.

BTW, I know I complained about the name `opSwitch` but we do follow that convention a bit with the OpDetauils constructors, `opBranch` and `opDefault`. Maybe we should rename them all at once - `detSwitch`, `detBranch`, etc.

Reply:
Oh, actually you want ops.pending.Len()+2 here, because you already added the byte for the opcode itself to ops.pending.